### PR TITLE
Closes AgileVentures/MetPlus_tracker#443

### PR DESCRIPTION
**Pull request by joaopapereira**

### CODE REVIEW

File: spec/controllers/tasks_controller_spec.rb [hunk 1]
```diff
@@ -255,70 +270,130 @@
...
+    describe 'Unauthorized access' do
+      let(:company) {FactoryGirl.create(:company)}
+      let(:company_contact) {FactoryGirl.create(:company_contact, :company => company)}
+      let(:job_developer) {FactoryGirl.create(:job_developer, :agency => agency)}
+      let(:case_manager) {FactoryGirl.create(:case_manager, :agency => agency)}
+      let(:job_seeker) {FactoryGirl.create(:job_seeker)}
+      let(:my_request) {xhr :get, :list_owners , {id: @task.id}, :format => :json}
       before :each do
-        @jd1 = FactoryGirl.create(:job_developer, :agency => agency)
         js = FactoryGirl.create(:job_seeker)
```
Comment (patmbolger) [inline spec/controllers/tasks_controller_spec.rb]: Don't need this object - use prior object.

File: spec/controllers/tasks_controller_spec.rb [hunk 2]
```diff
@@ -255,70 +270,130 @@
...
-        @task = Task.new_js_unassigned_cm_task js, agency
+        @task = Task.new_js_unassigned_jd_task js, agency
```
Comment (patmbolger) [inline spec/controllers/tasks_controller_spec.rb]: use `job_seeker` instead of `js`

File: app/policies/task_policy.rb [hunk 3]
```diff
@@ -13,6 +13,21 @@ def assign?
...
+    true
+  end
+
+  def list_owners?
+    return false if user.nil?
+    return true if not record.owner_agency.nil? and user.is_agency_admin?(record.owner_agency)
+    return true if not record.owner_company.nil? and user.is_company_admin?(record.owner_company)
+    false
   end
 end
```
Comment (patmbolger) [inline app/policies/task_policy.rb]: In the assign? method authorization is _not_ limited to agency or company admins, but the auth matrix shows this restriction.

Comment (joaopapereira) [inline app/policies/task_policy.rb]: Not following

Comment (joaopapereira) [inline app/policies/task_policy.rb]: After talking in the standup we decided to keep this as is and we will see what Chet needs when we finish MVP

**Pull request merged**